src/source.rs

```diff
@@ -12,23 +14,65 @@ pub trait Source: Debug {
 
     /// Collect all configuration properties available from this source and return
     /// a HashMap.
-    fn collect(&self) -> Result<HashMap<String, Value>>;
+    fn collect(&self) -> Result<HashMap<String, Value>, ConfigError>;
```

Comment:
Any particular reason why you don't use the `crate::error::Result<_>` anymore? If no, please revert this (and all similar) changes, so we don't have to specify `ConfigError` all the time in this module!

Reply:
I saw that `Result` is not exported from the crate, which I remember vaguely did not play nicely with rust-analyzer that automatically used crate's method signature when asked to "implement trait functions". And since it wasn't exported, it didn't compile, I had to think about why. It was a long time ago, so it might be fixed now.

And personally, I am not a big fan of exporting Result, because in the case of import `use config::*` it overwrites std's `Result` and it is annoying. Hence the decision to change it here, since people have to implement it.

I am open to discussion though, it might be good idea to test it once more as well.